pyiron/sphinx/base.py

```diff
@@ -713,128 +981,200 @@ def _set_kpoints(
-            )
-            self.input_writer.write_control(
-                file_name="control.sx",
-                cwd=self.working_directory,
-                control_str=self._control_str,
-            )
-            self.input_writer.write_basis(
-                file_name="basis.sx",
-                cwd=self.working_directory,
-                basis_str=self._basis_str,
-                save_memory=save_memory,
-                kpoints_odict=self._kpoints_odict
-            )
-            self.input_writer.write_hamilton(
-                file_name="hamilton.sx",
-                cwd=self.working_directory,
-                hamilston_str=self._hamilston_str,
-            )
+                self.input["EmptyStates"] = int(len(self.structure) + 3)
+        self.input_writer.structure = self.structure
+
+        self.set_structure_group()
+
+        if self.input["VaspPot"]:
+            potformat = "VASP"
+        else:
+            potformat = "JTH"
+
+        self.set_species_group(
+            check_overlap=self.input["CheckOverlap"],
+            potformat=potformat,
+        )
+
+        self.input.basis = Group({
+            "eCut": "EnCut/13.606",
+            "kPoint": Group({
+                "coords": "KpointCoords",
+                "weight": 1,
+                "relative": True
+            }),
+            "folding": "KpointFolding",
+        })
+        if self.input["SaveMemory"] is True:
+            self.input.basis.set_flag("saveMemory")
+
+        self.set_guess_group()
+
+        self.input.hamilton = Group({
+            "nEmptyStates": "EmptyStates",
+            "ekt": "Sigma",
+            "xc": "Xcorr"
+        })
+        if self._spin_enabled:
+            self.input.hamilton.set_flag("spinPolarized")
+
+        self.set_main_group()
+        if self.input["WriteWaves"] is False:
+            self.input.main.set_flag("noWavesStorage")
+
+
+    def write_wrapper(self, file_name="input.sx", enable_kjxc=False):
+        """
+        Write the main Sphinx script named input.sx.
+
+        Args:
+            file_name (str): name of the file to be written (optional)
+            enable_kjxc (bool): Whether or not to enable kjxc in pawPot.
+        """
+        file_name = posixpath.join(self.working_directory, file_name)
+        with open(file_name, "w") as f:
+            f.write(f"//{self.job_name}\n")
+            f.write("//SPHInX input file generated by pyiron\n")
+            f.write("format paw;\n")
+            f.write("include <parameters.sx>;\n")
+            f.write("include <userparameters.sx>;\n")
+            f.write("pawPot { include <pawPot.sx>; ")
+            if enable_kjxc:
+                f.write("kjxc; ")
+            f.write("}\n")
+            f.write("structure { include <structure.sx>; }\n")
+            f.write("basis { include <basis.sx>; }\n")
+            f.write("PAWHamiltonian { include <PAWHamiltonian.sx>; }\n")
+            f.write("initialGuess { include <initialGuess.sx>; }\n")
```

Comment:
No I also wanted to do this but thought maybe @samwaseda had a specific reason to parse everything to separate files (?) and didn't want to mess with that too much. But a single file would make even more sense if we decide to represent the Input object itself as a group. Not to mention kinder to our file quotas...

Reply:
We agreed in today's discussion to postpone this.

Reply:
Did we? I had the impression that we agreed that we'd write everything in one file in the end.

Reply:
I had also the impression that we postponed it...

Reply:
oh yes, Sam, you are right. What Michael suggested to postpone was to change the entire input into one Group(). But I agree that even with separate Group() objects at the pyiron level, the output could (and should) still be a single file.
I reopened the conversation.

Reply:
> I had the impression that we agreed that we'd write everything in one file in the end.
But still not the structure, right?

Reply:
I tend to agree on this, but don't have a strong opinion. On the plus side: structure can produce many lines difficult to parse by human eye, and "mistakes" in the structure are typically not correlated with mistakes elsewhere in the input file, so a simultaneous view of structure + other stuff is typically not needed.
I think this is something that you can rightfully decide on your own, but maybe add a short comment on your thoughts on this within the code to make sure that the next person does not change this again without considering your points.

Reply:
> But still not the structure, right?

I'm perfectly fine with or without structure. I'll gladly accept your choice, even if you put one of them without comments, because as far as I can see, it won't be a problem even if someone changes it again later (i.e. I don't think there'd be a question of backward compatibility).